includes/class-newspack-segments-model.php

```diff
@@ -381,9 +381,80 @@ function( $segment ) {
 			);
 		}
 
+		// Compute a criteria hash, so duplicate criteria sets can be found.
+		$segments = array_map(
+			function( $segment ) {
+				$segment['criteria_hash'] = md5( wp_json_encode( $segment['criteria'] ) );
+				return $segment;
+			},
+			$segments
+		);
+		// Mark segments as duplicate if any of the preceding (in priority) segments have the same criteria hash.
+		$segments = array_map(
+			function( $segment ) use ( $segments ) {
+				$preceding_segments                = array_filter(
+					$segments,
+					function( $s ) use ( $segment ) {
+						return $s['priority'] < $segment['priority'];
+					}
+				);
+				$segment['is_criteria_duplicated'] = count(
+					array_filter(
+						$preceding_segments,
+						function( $s ) use ( $segment ) {
+							return $s['criteria_hash'] === $segment['criteria_hash'];
+						}
+					)
+				) >= 1;
+				return $segment;
+			},
+			$segments
+		);
+
 		return $segments;
 	}
 
+	/**
+	 * Get the ids of prompts assigned to a segment.
+	 *
+	 * @param int $segment_id The segment ID.
+	 */
+	private static function get_segments_prompts_ids( $segment_id ) {
```

Comment:
nit: should it be get_segment (singular) _prompts_ids ?

Reply:
It was meant as "segment's prompts ids", but yeah, that's not super clear. Updated in c9e36e5674cf6d3648836337c699031070f95deb